File: src/BookReader.UniApp/models/ReadingProgress.js
export default class ReadingProgress {
  constructor(bookId, chapterId, position, timestamp) {
    this.bookId = bookId;           // 书籍ID
    this.chapterId = chapterId;     // 章节ID
    this.position = position;        // 阅读位置（百分比）
    this.timestamp = timestamp;      // 最后阅读时间
  }
}

File: src/BookReader.API/Models/ReadingProgress.cs
namespace BookReader.API.Models;

public class ReadingProgress
{
    public string BookId { get; set; } = null!;
    public int ChapterIndex { get; set; }
    public float Position { get; set; }  // 章节内的阅读位置（0-1）
    public DateTime LastReadTime { get; set; }
    public ReaderSettings Settings { get; set; } = new();
}
